test/functional/wallet_address_types.py

```diff
@@ -95,34 +96,42 @@ def get_balances(self, confirmed=True):
         else:
             return [self.nodes[i].getunconfirmedbalance() for i in range(4)]
 
+    def remove_origin(self, desc):
+        return re.sub('\[.*?\]', '', desc)
+
     def test_address(self, node, address, multisig, typ):
         """Run sanity checks on an address."""
         info = self.nodes[node].getaddressinfo(address)
+        assert('descriptor' in info)
         assert(self.nodes[node].validateaddress(address)['isvalid'])
         if not multisig and typ == 'legacy':
             # P2PKH
             assert(not info['isscript'])
             assert(not info['iswitness'])
             assert('pubkey' in info)
+            assert_equal(self.remove_origin(info['descriptor']), "pkh(%s)" % info['pubkey'])
```

Comment:
It seems like it'd be good to check inferred origins instead of skipping them. If it isn't easy to do this, maybe you could add a comment to the test saying why.

Reply:
Done, rewritten the test to compare with information put in PSBT (which also reports fingerprint and path in `decodepsbt`).